### PR TITLE
Add support for TIFF files in load_file_headers

### PR DESCRIPTION
**Pull request by pierotofy**
Hello :hand:

This PR adds support for TIFF files in the `load_file_headers`, which is currently missing.

Hope this is useful for others. :+1:

Please let me know if changes are needed.

### CODE REVIEW

File: libs/mve/image_io.cc [hunk 1]
```diff
@@ -628,6 +635,104 @@ tiff_error_handler (char const* /*module*/, char const* fmt, va_list ap)
...
 
+ImageHeaders
+load_tiff_file_headers (std::string const& filename)
+{
+    ImageHeaders headers;
+    TIFFSetWarningHandler(nullptr);
+    TIFFSetErrorHandler(tiff_error_handler);
+
+    TIFF* tif = TIFFOpen(filename.c_str(), "r");
+    if (!tif)
```
Comment (simonfuhrmann) [inline libs/mve/image_io.cc]: `if (tif == nullptr)`

File: libs/mve/image_io.cc [hunk 2]
```diff
@@ -628,6 +635,104 @@ tiff_error_handler (char const* /*module*/, char const* fmt, va_list ap)
...
+        TIFFGetField(tif, TIFFTAG_SAMPLESPERPIXEL, &channels);
+        TIFFGetField(tif, TIFFTAG_BITSPERSAMPLE, &bits);
+        TIFFGetField(tif, TIFFTAG_SAMPLEFORMAT, &sampleFormat);
+
+        headers.width = width;
+        headers.height = height;
+        headers.channels = channels;
+        headers.type = IMAGE_TYPE_UNKNOWN;
+
+        if (bits == 8){
```
Comment (simonfuhrmann) [inline libs/mve/image_io.cc]: space before `{` here and below

File: libs/mve/image_io.cc [hunk 3]
```diff
@@ -628,6 +635,104 @@ tiff_error_handler (char const* /*module*/, char const* fmt, va_list ap)
...
+                case SAMPLEFORMAT_INT:
+                    headers.type = IMAGE_TYPE_SINT8;
+                    break;
+                case SAMPLEFORMAT_IEEEFP:
+                    headers.type = IMAGE_TYPE_FLOAT;
+                    break;
+                default:
+                    break;
+            }
+        }else if (bits == 16){
```
Comment (simonfuhrmann) [inline libs/mve/image_io.cc]: space after `}` here and below

File: libs/mve/image_io.cc [hunk 4]
```diff
@@ -628,6 +635,104 @@ tiff_error_handler (char const* /*module*/, char const* fmt, va_list ap)
...
+ImageHeaders
+load_tiff_file_headers (std::string const& filename)
+{
+    ImageHeaders headers;
+    TIFFSetWarningHandler(nullptr);
+    TIFFSetErrorHandler(tiff_error_handler);
+
+    TIFF* tif = TIFFOpen(filename.c_str(), "r");
+    if (!tif)
+        throw util::Exception("TIFF file format not recognized");
```
Comment (simonfuhrmann) [inline libs/mve/image_io.cc]: Would that be a `util::FileException`?

File: libs/mve/image_io.cc (hunk 1 above)
Comment (pierotofy) [reply libs/mve/image_io.cc]: Better, yes. Thank you.

File: libs/mve/image_io.cc (hunk 4 above)
Comment (pierotofy) [reply libs/mve/image_io.cc]: Good catch, yes it would be better instead of the generic one. :+1:

**Comment by pierotofy** [comment]
Much appreciate the feedback!

**Pull request merged**